test/debug/info_test.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative '../support/test_case'
+
+module DEBUGGER__
+  class BasicInfoTest < TestCase
+    def program
+      <<~RUBY
+     1| def foo
+     2|   @var = 10
+     3|   a = 1
+     4|   @var + 1
+     5| end
+     6|
+     7| foo
+      RUBY
+    end
+
+    def test_info_prints_locals_by_default
+      debug_code(program) do
+        type 'b 5'
+        type 'c'
+        type 'info'
+        assert_line_text(
+" %self => main\n" \
+" %return => 11\n" \
+" a => 1\n" \
+" @var => 10\n"
+        )
+        type 'q!'
+      end
+    end
+  end
+
+  class InfoThreadsTest < TestCase
+    def program
+      <<~RUBY
+       1| def foo
+       2|   Thread.new { sleep 30 }
+       3| end
+       4|
+       5| foo
+       6| "placeholder"
+      RUBY
+    end
+
+    def test_prints_current_thread
+      debug_code(program) do
+        type 'b 6'
+        type 'c'
+        type 'info threads'
+        assert_line_text(/#0 \(sleep\)@.*:6:in `<main>'/)
+        type 'q!'
+      end
+    end
+
+    def test_prints_the_other_thread
+      debug_code(program) do
+        type 'b 6'
+        type 'c'
+        type 'info threads'
+        assert_line_text(/#1 \(sleep\)@.*:2 sleep_forever/)
```

Comment:
@ko1 this test case only fails in the Ruby 2.7 test suite and it's not reproduceable to me.

```
--> #0 (sleep)@/tmp/debugger20210608-1615-18rjf3w.rb:6:in `<main>'
The following threads are not managed yet by the debugger:     #<Thread:0x000056409a6dbc40 /tmp/debugger20210608-1615-18rjf3w.rb:2 run>"
```

I'm not sure how a thread can escape the `thread_begin` tp event. or maybe it's registered incorrectly...? 
anyway, I think I'm not able to diagnose the issue, can you give it a look? thanks

Reply:
Thank you. Tests help us.

Reply:
Could you try on master?
I didn't check it details, but it can be fixed (and it can be not fixed.

Reply:
I have rebased it on master but it still fails on 2.7 😕

Reply:
Thank you I'll check it.

Reply:
@ko1 (part of) the cause seems to be a race condition between the test thread's (the one in `foo`) registration  (`#thread_client_create`) and the `info threads` command. 

Currently, it's fixed by postponing the `info` call a bit.